networkit/cpp/centrality/GroupClosenessLocalSearch.cpp

```diff
@@ -0,0 +1,1143 @@
+        runSanityCheck();
+#endif // NETWORKIT_SANITY_CHECKS
+
+        // Approximate the farness decrease for each vertex by estimating their BFS DAG
+        // size
+        estimateFarnessDecrease();
+        candidatesToAdd.build_heap(candidatesToAddVec.begin(), candidatesToAddVec.end());
+
+        std::atomic<bool> stop{false};
+        std::atomic<int> threadToSelect{-1};
+
+#pragma omp parallel
+        {
+            while (!stop.load(std::memory_order_relaxed)) {
+
+                node v = none;
+#pragma omp critical
+                {
+                    if (candidatesToAdd.empty()) {
+                        stop.store(true, std::memory_order_relaxed);
+                    } else {
+                        v = candidatesToAdd.extract_top();
+                        if (approxFarnessDecrease[v] <= 0.f) {
+                            stop.store(true, std::memory_order_relaxed);
+                            v = none;
+                        }
+                    }
+                }
+
+                if (v == none)
+                    break;
+
+                copyStatusToThread();
+#ifdef NETWORKIT_SANITY_CHECKS
+                runSanityCheck(none, true);
+#endif // NETWORKIT_SANITY_CHECKS
+
+                const WeightType farnessDecrease = computeFarnessDecrease(v, increaseU, true, true);
+
+#ifdef NETWORKIT_SANITY_CHECKS
+                if (farnessDecrease != invalidDecr)
+                    runSanityCheck(v, true);
+#endif // NETWORKIT_SANITY_CHECKS
+
+                if (farnessDecrease == invalidDecr
+                    || insuffcientDecrease(farnessDecrease, increaseU))
+                    continue;
+
+                stop.store(true, std::memory_order_relaxed);
+                threadToSelect.store(omp_get_thread_num(), std::memory_order_relaxed);
+
+                decreasePerThread[omp_get_thread_num()] = std::make_pair(v, farnessDecrease);
+            }
+        }
+
+        const int thread = threadToSelect.load(std::memory_order_relaxed);
+
+        if (thread != -1) { // There a convenient swap
+            node v;
+            WeightType farnessDecrease;
+            std::tie(v, farnessDecrease) = decreasePerThread[thread];
+
+            // Removing u from group
+            groupFarness -= (farnessDecrease - increaseU);
+
+            // Put v into the group
+            group.insert(v);
+
+            storeStatusFromThread(thread);
+#ifdef NETWORKIT_SANITY_CHECKS
+            runSanityCheck();
+#endif // NETWORKIT_SANITY_CHECKS
+            return true;
+        }
+
+        // Swaps with u did not improve the closeness enough, restore u into the group
+        group.insert(u);
+        const WeightType decreaseU = computeFarnessDecrease(u);
+        tlx::unused(decreaseU);
+#ifndef NDEBUG
+        if (G->isWeighted())
+            assert(std::abs(static_cast<double>(decreaseU - increaseU)) <= 1e-9);
+        else
+            assert(decreaseU == increaseU);
+#endif // NDEBUG
```

Comment:
Wrapping these statements in `#ifndef NDEBUG` is unnecessary, isn't it? Surely the compiler will be able to optimize them away if `NDEBUG` is set? (Unless you're doing more complicated stuff than just `assert`ing something.)